cli/find.php

```diff
@@ -111,27 +108,53 @@
     fputcsv($fp, ['Table', 'Column', 'courseid', 'idnumber']);
 }
 
-// Output the result.
-foreach ($result as $table => $columns) {
-    foreach ($columns as $column => $rows) {
+// Perform the search.
+$searchlist = helper::build_searching_list($tables);
+
+// Output the result for each table.
+foreach ($searchlist as $table => $columns) {
+    // Summary mode.
+    $limit = $options['summary'] ? 1 : 0;
+
+    foreach ($columns as $column) {
+        // Show the table and column being searched.
+        $colname = $column->name;
+        echo "Searching in table: $table, column: $colname\n";
+
+        // Perform the search.
+        if (!empty($options['regex-match'])) {
+            $result = helper::regular_expression_search($search, $table, $column, $limit);
+        } else {
+            $result = helper::plain_text_search($search, $table, $column, $limit);
+        }
+
+        // Notifying the user if no results were found.
+        if (empty($result)) {
+            echo "No results found.\n";
+            continue;
+        } else {
+            echo count($result) . " results found.\n";
+        }
+
+        $rows = reset($result)[$colname];
         if ($options['summary']) {
             $courseid = reset($rows)->courseid ?? '';
             $courseidnumber = reset($rows)->courseidnumber ?? '';
-            fputcsv($fp, [$table, $column, $courseid, $courseidnumber]);
+            fputcsv($fp, [$table, $colname, $courseid, $courseidnumber]);
```

Comment:
the header should be written earlier, if the file is empty we still should end up with a file with 1 header row and nothing else